partitura/io/importmatch.py

```diff
@@ -849,5 +852,66 @@ def add_staffs(part: Part, split: int = 55, only_missing: bool = True) -> None:
     part.add(score.Clef(staff=2, sign="F", line=4, octave_change=0), 0)
 
 
+def validate_match_ids(mf):
+    """
+    Check a matchfile for duplicate snote and note IDs.
+
+    This function will:
+    - remove all deletions with a score ID that occurs in multiple lines.
+    - remove all insertions with a performance ID that occurs in multiple lines.
+
+    Handles cases with conflicting match/insertion(s) and match/deletion(s) tuples 
+    with any number of insertions or deletions and a single match by keeping 
+    only the match.
+
+    Unhandled cases: 
+    - multiple conflicting matches: all are kept.
+    - multiple insertions with the same ID: all are deleted.
+    - multiple deletions with the same ID: all are deleted.
+
+    Parameters
+    ----------
+    mf: MatchFile
+        MatchFile to validate
+
+    Returns
+    -------
+    Updates the representation of the matchfile by removing match lines.
+    """
+
+    # Check if the matchfile is valid (i.e. check for snote duplicates)
+    sids = np.array([n.Anchor for n in mf.snotes])
+    # First check if score ids are unique
+    sids_unique, counts = np.unique(sids, return_counts=True)
+    sids_to_check = sids_unique[np.where(counts > 1)[0]]
+    if len(sids_to_check) > 0:
+        indices_to_del = []
+        for i, line in enumerate(mf.lines):
+            if isinstance(line, BaseDeletionLine):
```

Comment:
While the limitations of the current implementation are explicitly stated in the documentation. I'm not sure if that would break something when computing features. If addressing this issue takes a lot of time, maybe we can have just a test where we make sure that nothing goes wrong when parsing a match file with multiple deletions (without at least one corresponding matched score note) and multiple insertions (without a corresponding matched score note).